src/components/PopupWithForm.js

```diff
@@ -7,6 +7,7 @@ class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form.form");
     this._handleFormSubmit = handleFormSubmit; // a callback function pass as an argument
     this._inputList = this._popupForm.querySelectorAll(".form__input");
+    this._submitButton = this._popupForm.querySelector(".modal__button");
   }
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`